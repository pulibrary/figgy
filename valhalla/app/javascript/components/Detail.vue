<template>
  <div class="sidebar">
    <div id="detail" class="actions">
      <img id="detail_img" :src="selectedImageUrl"></img>
    </div>
  </div>
</template>

<script>
export default {
  name: 'detail',
  computed: {
    selected: {
      get () {
        return this.$store.state.selected
      }
    },
    selectedImageUrl: function () {
      return this.selected.length === 1 ? this.selected[0].url : ''
    }
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
#detail {
  height: 100%;
  padding: 15px;
}

#detail img {
  height: 100%;
}
</style>
