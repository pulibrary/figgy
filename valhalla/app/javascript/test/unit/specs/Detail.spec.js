import Vue from 'vue'
import Vuex from 'vuex'
import { shallow, createLocalVue } from 'vue-test-utils'
import Detail from '@/components/Detail'
const localVue = createLocalVue()
localVue.use(Vuex)

describe('Detail.vue', () => {
  let cmp

  beforeEach(() => {
    cmp = shallow(Detail, {
      computed: {
        selected: {
          get () {
            return [
              {
                "id": "50b5e49b-ade7-4278-8265-4f72081f26a5",
                "url": "http://localhost:3000/image-service/50b5e49b-ade7-4278-8265-4f72081f26a5/full/400,/0/default.jpg",
                "label": "1",
                "pageType": "single"
              }
            ]
          }
        }
      },
      localVue
    })
  })

  it('renders a $store.state value return from computed', () => {
    expect(cmp.find('#detail_img').html()).toEqual('<img id="detail_img" src="http://localhost:3000/image-service/50b5e49b-ade7-4278-8265-4f72081f26a5/full/400,/0/default.jpg">')
  })

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot()
  })

  it('selectedImageUrl function has the expected value', () => {
    expect(cmp.vm.selectedImageUrl).toEqual("http://localhost:3000/image-service/50b5e49b-ade7-4278-8265-4f72081f26a5/full/400,/0/default.jpg")
  })

})
