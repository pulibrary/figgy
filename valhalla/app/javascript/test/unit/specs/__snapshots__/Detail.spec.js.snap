// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail.vue has the expected html structure 1`] = `
<div
  class="sidebar"
>
  <div
    class="actions"
    id="detail"
  >
    <img
      id="detail_img"
      src="http://localhost:3000/image-service/50b5e49b-ade7-4278-8265-4f72081f26a5/full/400,/0/default.jpg"
    />
  </div>
</div>
`;
